### ISSUE TITLE
Related Works

### ISSUE THREAD

**Issue by Jeremy-Siburian** [Jeremy-Siburian/graduation-thesis-research#1]
Papers or datasets related to robotic slip detection using distributed 3-axis tactile sensing

**Comment by Jeremy-Siburian** [comment]
# Leveraging distributed contact force measurements for slip detection: a physics-based approach enabled by a data-driven tactile sensor

- Venue: ICRA 2022
- Paper URL: https://arxiv.org/abs/2109.11504
- Dataset: None

## Summary

- Method: Physics-based, Coulomb model and Stick Ratio (SR)
- Sensor: Vision-based

## Motivation 

- Vast majority of works purely based on distributed measurements have focused on data-driven approaches.
- Model-based approach to bypass dependency on training data while retaining reliable prediction accuracy.
- Novelty point: Distributed contact force measuremens are leveraged to measure the slipping area

## Methods

### Tactile Sensor

- Tactile sensor used: https://www.mdpi.com/1424-8220/19/4/928
- Camera frames -> DNN -> discretized force distribution
- 20x20, 400 measurement points with xyz axes measurements
- Only one sensor used for computational efficiency

### Method

#### Baseline 

- Dry friction law (Coulomb friction law)
- Computed as total quantities (summations) over the entire contact region

#### Stick Ratio Approach

- Based on the concept of incipient slip (relative displacement occurs only in a narrow region of the contact interface, preceding gross slip)
- Consider each taxel as a discretized contact point, compare normal and shear forces at a single point
- Ratio of stick area (C) against contact area (A) is called stick ratio (SR)

## Experiment

- Constant value of friction coefficient set to 0.45 (assumption from previous work)
- Grasping sequence defined a priori for each test object, initial object pose facilitated successful grasp
- External load manually applied until slip happened
- Change of slip state (STICK or SLIP) recorded manually, ground truth labels to compute performance scores
- Label data is for evaluation only (model-based), not used for training
- Tested on 5 objects, 5 displacements = 4 translational (2 horizontal 2 vertical), 1 rotational with 90 angle

### Experimental Results

- Evaluation metrics: Accuracy, Precision, Recall
- Results were obtained as average of three different runs for each experiment (reduce uncertainty of ground truth label)
- Stick ratio outperforms baseline strategy, able to handle rotational slip
- Slip = stick ratio < threshold of 0.5 (arbitrarily chosen)

## Limitations

- Performance gap between translational and rotational slip -> marginal drop in accuracy of contact forces during rotational motion -> geometric interferene between the slider and the edges of the squared tactile sensor when rotating object
- Possible solution: dome-shaped tactile sensor?

## Comments

- Should be straight forward to implement onto uSkin
- Only uses one sensor, limited frequency
- Still relies on threshold and assumption of friction coefficient 
- Limited objects and limited testing scenario (understandable because we need to easily obtain ground truth labels)
- Would be interesting to test onto Lorenzo's dataset for example (pick-and-place)

**Comment by Jeremy-Siburian** [comment]
# Slip Detection for Grasp Stabilization With a Multifingered Tactile Robot Hand

Venue: IEEE Transactions On Robotics (2021)

## Summary

- Sensor: TacTip (Optical biomimetic)
- Slip Method: 1 model-based + 2 learning-based (SVM & LogReg)

### Main Contributions

- Compare and understand simple slip detection methods based on pin velocity features 
- Test multiple slip scenarios with 11 different objects
- Testing two real-world scenarios: adding weight to destabilize a grasp & perform first-time grasping

## Motivation

- Debate on model-based versus model-free is far from being settled, and merits of both approaches should be investigated.

## Method

Compare three binary classifiers:
- A threshold on the magnitude of mean pin velocity
- SVM applied to either linear or nonlinear combinations of pin velocities
- logistic regression (LogReg) applied to individual pin velocities

Ground truth data:
- Data labeling: ArUco optical marker on the slipping object and recorded its height with another JeVois camera
- Reasoning: labeling process autonomous and contains no labeling bias

Note: 
- For efficiency, they test the effect of downsampling these data
- Vary the number of frames in the slip class and randomly downsample the data before optimizing each classifier

## Experiment (Single Finger)

Uses a rail system and single tactile fingertip
Start with offline performance with pre-collected data
Online performance to detect slip in real-time (best classifier only)

### Offline Performance
#### Threshold Method
- Threshold was chosen by testing 10 values (Best performing was 0.14 pixels/s)
- F1 score = 0.753
- Significantly worse than SVM and LogReg
- Suffers from not rejecting false positives (it does not distinguish slip from changes of pin position unrelated to slip)

#### SVM and LogReg methods
- SVM outperforms LogReg across all tests
- Reaction time: able to classify in sub-milisecond time (therefore we can select purely on score)

### Online Performance
- Test the ability to detect slip in real-time (best classifier only) -> action to minimize slip
- Two reaction strategies: any frame vs. two consecutive frames (latter is to reduce false positives at the cost of reaction time)
- Result: Similar task success, but FP rate drops from 20% to 0% when using two consecutive frames

## Experiment (Whole Hand Testing)

- TO-DO READ

### Evaluation

- Does not use normal method of using accuracy (Reasoning: usually more examples of nonslip than slip, but they do not want to discard data that could give a better classifier, therefore normal accuracy is not a good method)
- F1 score is used (which takes into account the precision and recall)

## Comments

- Paper provides a very clear and comprehensive comparison of the slip methods, great paper overall

Interesting things to adopt: 
- Compare performance of using one vs. two sensors (number of sensors detecting slip)
- Number of consecutive frames detecting slip (vary the frames, can even test with the threshold method used in Fuso internship)

## Interesting Reference to Read

- [ ] L. Li, F. Sun, B. Fang, Z. Huang, C. Yang, and M. Jing, “Learning to detect slip for stable grasping,” in Proc. IEEE Int. Conf. Robot. Biomimetics, 2017, pp. 430–435
- [ ] M. Li, Y. Bekiroglu, D. Kragic, and A. Billard, “Learning of grasp adaptation through experience and tactile sensing,” in Proc. IEEE/RSJ Int. Conf. Intell. Robots Syst., Sep. 2014, pp. 3339–3346
